### PR TITLE
cloud-utils: 0.29 -> 0.30

### PR DESCRIPTION
**Pull request by Mic92**
###### Motivation for this change

The patch I removed was actually not used by this version anymore.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/virtualisation/grow-partition.nix [hunk 1]
```diff
@@ -24,7 +24,12 @@ with lib;
       copy_bin_and_libs ${pkgs.gnused}/bin/sed
       copy_bin_and_libs ${pkgs.utillinux}/sbin/sfdisk
       copy_bin_and_libs ${pkgs.utillinux}/sbin/lsblk
-      cp -v ${pkgs.cloud-utils}/bin/.growpart-wrapped $out/bin/growpart
+
+      substitute "${pkgs.cloud-utils}/bin/.growpart-wrapped" "$out/bin/growpart" \
+        --replace "${pkgs.bash}/bin/sh" "/bin/sh" \
+        --replace "awk" "gawk" \
+        --replace "sed" "gnused"
```
Comment (orivej) [inline nixos/modules/virtualisation/grow-partition.nix]: This looks strange: you copy `bin/sed` and then replace sed with gnused. And why don't we copy `bin/awk`?

Comment (Mic92) [reply nixos/modules/virtualisation/grow-partition.nix]: I guess this is a name collision with busybox.

Comment (Mic92) [reply nixos/modules/virtualisation/grow-partition.nix]: I actually only restored the previous behavior. Do we have any name collision with gawk and busybox?

**Comment by orivej** [comment]
`nixos/tests/cloud-init.nix` does not test that `growpart` really works because it boots a VM without partitions, so I'm not very confident.

**Comment by Mic92** [comment]
Checkout `nix-build nixos/release-combined.nix -A nixos.tests.ec2-nixops` as recommend in cloud-tools package, it does the check.

**Review by orivej — APPROVED** [review]

File: nixos/modules/virtualisation/grow-partition.nix (hunk 1 above)
Comment (copumpkin) [reply nixos/modules/virtualisation/grow-partition.nix]: Super super weird to modify a store path in place taking advantage of the fact that you're in the initrd 😱

Comment (copumpkin) [reply nixos/modules/virtualisation/grow-partition.nix]: Never mind, it's not in place, but still weird to patch the wrapper when we own it.

**Pull request merged**